src/Utils/Utils.jl

```diff
@@ -218,32 +218,26 @@ INPUT:
 - ``ϕ``    -- a matrix, which can be either dense or sparse
 - ``name`` -- the name of the matrix (for the output message)
 """
-function print_sparsity(ϕ::AbstractVector{Float64}, name::String="")
+function print_sparsity(ϕ::Union{AbstractVector{Float64}, SparseMatrixExp{Float64}}, name::String="")
```

Comment:
it's not related to this issue, but shouldn't this be  a `ϕ::Union{AbstractMatrix{Float64}, SparseMatrixExp{Float64}}`?

Reply:
Agreed, let me fix this in `master` directly.

Reply:
Actually, no, there is a second function for this below :)

Reply:
we won't have a matrix given as `AbstractVector{Float64}` as input, or if it does it cannot make sense of `ϕ[(2*bi-1):(2*bi), (2*bj-1):(2*bj)]`

Reply:
shall we fix it here or outside this issue?

Reply:
Do you mean it should be an `AbstractMatrix`?

Reply:
yes

Reply:
outside

Reply:
ok, then let me merge this please.